src/App.jsx

```diff
@@ -3,69 +3,134 @@ import './App.css'
     const apiKey = import.meta.env.VITE_API_KEY;
     let response = {};
+    let url = '';
 
     switch (fxn) {
       case 'now_playing':
-        response = await fetch(`https://api.themoviedb.org/3/movie/now_playing?language=en-US&page=${pages}&api_key=${apiKey}`);
+        response = await fetch(`https://api.themoviedb.org/3/movie/now_playing?language=en-US&page=${page}&api_key=${apiKey}`);
         break;
       case 'search':
         response = await fetch(`https://api.themoviedb.org/3/search/movie?query=${searchTerm}&include_adult=false&language=en-US&page=1&api_key=${apiKey}`);
         break;
-      default: response = {};
-
+      case 'by_genre':
+        response = await fetch(`https://api.themoviedb.org/3/discover/movie?with_genres=${genre}&language=en-US&page=${page}&api_key=${apiKey}`);
+        break;
+      case 'sort_by_attribute':
+        if (genre === ''){
+          url = `https://api.themoviedb.org/3/discover/movie?include_adult=false&include_video=true&language=en-US&page=${page}&sort_by=${sortAttribute}&api_key=${apiKey}`;
+        } else {
+          url = `https://api.themoviedb.org/3/discover/movie?include_adult=false&with_genres=${genre}&language=en-US&page=${page}&sort_by=${sortAttribute}&api_key=${apiKey}`;
+        }
+        response = await fetch(url);
+        break;
+      default:
+        response = response = await fetch(`https://api.themoviedb.org/3/movie/now_playing?language=en-US&${page}=1&api_key=${apiKey}`);
     }
 
     const data = await response.json();
     setLoadedMovies(prevMovieList.concat(data.results));
   }
 
   useEffect(() => {
-    generalFetchMovies('search', [], 1, searchTerm);
-  }, [searchTerm]);
-
-  useEffect(() => {
-    generalFetchMovies('now_playing', loadedMovies, page, '');
-  }, [page]);
+    generalFetchMovies(apiFunctionCall, loadedMovies, page, searchTerm, genre, sortAttribute);
+  }, [searchTerm, page, genre, sortAttribute]);
 
   const loadAdditionalPage = () => {
     setPage(page + 1);
-    generalFetchMovies('now_playing', loadedMovies, page+1, '');
   }
 
   const onSubmitSearch = (searchTerm) => {
     setSearchTerm(searchTerm);
+    setApiFunctionCall('search');
+    setLoadedMovies([]);
+    setPage(1);
   }
 
   const goToSearchView = () => {
     setSearchTerm('');
+    setApiFunctionCall('search');
     setLoadedMovies([]);
+    setPage(1);
   }
 
   const goToNowShowingView = () => {
     setPage(1);
-    generalFetchMovies('now_playing', [], 1, '');
+    setLoadedMovies([]);
+    setApiFunctionCall('now_playing');
+    generalFetchMovies('now_playing', [], 1, '', '', '');
+  }
+
+  const getNewGenre = (newGenreName) => {
+    setLoadedMovies([]);
+    setApiFunctionCall('by_genre');
+    setGenre(newGenreName);
+    setPage(1);
+  }
+
+  const sortByAttribute = (sortAttribute) => {
+    setLoadedMovies([]);
+    setApiFunctionCall('sort_by_attribute');
+    setSortAttribute(sortAttribute);
+    setPage(1);
+  }
```

Comment:
Did you notice that every time the `apiFunctionCall` changes, you reset values back to their default? This can be accomplished instead with `useEffect` (or might even be able to combine it into your effect above). 

```
useEffect(() => {
  setLoadedMovies([]);
  setPage(1);
}, [setLoadedMovies, setPage, apiFunctionCall])
```